crates/holochain_cascade/src/authority/get_agent_activity_query/hashes.rs

```diff
@@ -53,6 +53,7 @@ impl Query for GetAgentActivityHashesQuery {
                 -- is an action authored by this agent
                 Action.author = :author
                 AND DhtOp.type = :chain_op_type
+                AND DhtOp.withhold_publish IS NULL
```

Comment:
This field is only ever set in the authored database. So this is only a problem if the author ends up answering an activity query for themselves and the query falls back to the authored database. Still, it seems like a hole worth closing.

Reply:
note, https://github.com/holochain/holochain/pull/4295 may help with this kind of thing, ensuring that a query only runs over one of the ops databases

Reply:
That would certainly help with queries that are only supposed to run on one kind of database. This query lives in the cascade but it's actually the pattern from the `holochain_state` crate, which provides queries that can be run against cache/dht/authored databases.

Maybe it's not even valid that we're answering activity queries from our authored data? If we're acting as an authority for ourselves then we should probably only return data that has been placed in the DHT database. 

This is a bit of a general question for me - should we be able to describe this in general to the cascade? Some queries are valid to be run against any database and it's useful that if the DHT database is being rebuilt for some reason, that we can query our authored data. But there are maybe some cases like this where we should maybe not do it

Reply:
Never mind, this query is actually only called with a DHT database... So we're already doing that. Maybe this change isn't valid then and there's a different explanation for why agent activity queries to the author are resulting in countersigning sessions completing in Jost's test...

I'm going to put this back into draft for now and see if we can figure out why that's happening